src/libs/OptionsListUtils.ts

```diff
@@ -96,81 +154,80 @@ Onyx.connect({
     },
 });
 
-let allTransactions = {};
+let allTransactions: OnyxCollection<Transaction> = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.TRANSACTION,
     waitForCollectionCallback: true,
-    callback: (val) => {
-        if (!val) {
+    callback: (value) => {
+        if (!value) {
             return;
         }
-        allTransactions = _.pick(val, (transaction) => transaction);
+
+        allTransactions = Object.keys(value)
+            .filter((key) => !!value[key])
+            .reduce((result: OnyxCollection<Transaction>, key) => {
+                if (result) {
+                    // eslint-disable-next-line no-param-reassign
+                    result[key] = value[key];
+                }
+                return result;
+            }, {});
     },
 });
 
 /**
  * Adds expensify SMS domain (@expensify.sms) if login is a phone number and if it's not included yet
- *
- * @param {String} login
- * @return {String}
  */
-function addSMSDomainIfPhoneNumber(login) {
+function addSMSDomainIfPhoneNumber(login: string): string {
     const parsedPhoneNumber = parsePhoneNumber(login);
     if (parsedPhoneNumber.possible && !Str.isValidEmail(login)) {
-        return parsedPhoneNumber.number.e164 + CONST.SMS.DOMAIN;
+        return parsedPhoneNumber.number?.e164 + CONST.SMS.DOMAIN;
```

Comment:
Can number be `undefined` although it's `possible`?

Reply:
Based on this type provided from the `awsome-phonenumber` library, yes :smile:
<img width="500" alt="image" src="https://github.com/Expensify/App/assets/25230417/42e8b298-4cd7-4e58-925b-04674c33777d">